(GH-14) Cross Platform Task

louisfischer opened the pull request:
Fixes #14 

This pull request is to begin the process of creating a cross-platform agent for Azure DevOps that will run on Windows and Non-Windows Agent machines. Please review and let me know if this approach will work or what you would like to see changed. The original Powershell bootstrap is still there without any modifications.

Below is an image of the script running using a basic build.cake script

![image](https://user-images.githubusercontent.com/1905010/77028717-a8a2a200-6967-11ea-801d-9f932603d3b9.png)
![image](https://user-images.githubusercontent.com/1905010/77028751-bf48f900-6967-11ea-83c9-18db20006231.png)
![image](https://user-images.githubusercontent.com/1905010/77028764-ca9c2480-6967-11ea-9001-0f5d3ab9f4e5.png)

louisfischer commented:
@gep13 @devlead will one of you let me know if anything needs to be updated or reworked?

patriksvensson commented on package.json:
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "cake-task",
+  "version": "1.0.0",
```
Do we need to update this version before publishing?

louisfischer replied:
@patriksvensson modified the build.cake script to version package.json to be consistent with the other versioning locations

patriksvensson commented on src/cake.ts:
```diff
@@ -0,0 +1,76 @@
+
+    // Start Cake
+    console.log('Executing build script...');
+    const exitCode = await tl.exec(cakeToolPath, `"${$script}" -target="${$target}" -verbosity="${$verbosity}" --paths_tools="${toolPath}" ${$arguments}`);
```
Perhaps change `${toolPath}` to `${$toolPath}` so it's consistent?

louisfischer replied:
@patriksvensson done.

Commit:
Added CrossPlatform Script.


Added ignore files


Adjust build.cake to build xplat agent


Resolve absolute path based on script


Updated Cake.ts for toolPath and version for package.json

louisfischer commented:
@patriksvensson @gep13 @devlead Any other suggestions or feedback you may have for this?

gep13 committed:
(maint) Ignore .vs folder

This doesn't need to go into source control

gep13 committed:
(GH-14) Remove usage of PowerShell execution

gep13 committed:
(GH-14) Update Cake DAG

Ensure that ncc is installed before using it.

gep13 committed:
(GH-14) Default all values

Also ensure usage of -- in options.

gep13 committed:
(GH-14) Update file name

There is no cake.js file.  The cake.ts file is converted to an index.js
file.

gep13 reviewed (approved):
LGTM!

gep13 commented:
@louisfischer I made a few changes to this, but on the whole, this was a great PR, thank you very much for providing this!
